contracts/test/lib/PauseManager.ts

```diff
@@ -80,6 +81,58 @@ describe("PauseManager", () => {
     });
   });
 
+  describe("Updating pause type and unpausetype roles", () => {
+    it("should fail updatePauseTypeRole if unused pause type is used", async () => {
+      const updateCall = pauseManager.updatePauseTypeRole(UNUSED_PAUSE_TYPE, DEFAULT_ADMIN_ROLE);
+      await expectRevertWithCustomError(pauseManager, updateCall, "PauseTypeNotUsed");
+    });
+
+    it("should fail updateUnpauseTypeRole if unused pause type is used", async () => {
+      const updateCall = pauseManager.updateUnpauseTypeRole(UNUSED_PAUSE_TYPE, DEFAULT_ADMIN_ROLE);
+      await expectRevertWithCustomError(pauseManager, updateCall, "PauseTypeNotUsed");
+    });
+
+    it("should fail updatePauseTypeRole if correct role not used", async () => {
+      const updateCall = pauseManager.connect(nonManager).updatePauseTypeRole(GENERAL_PAUSE_TYPE, DEFAULT_ADMIN_ROLE);
+      await expectRevertWithReason(updateCall, buildAccessErrorMessage(nonManager, PAUSE_ALL_ROLE));
+    });
+
+    it("should fail updateUnpauseTypeRole if correct role not used", async () => {
+      const updateCall = pauseManager.connect(nonManager).updateUnpauseTypeRole(GENERAL_PAUSE_TYPE, DEFAULT_ADMIN_ROLE);
+      await expectRevertWithReason(updateCall, buildAccessErrorMessage(nonManager, UNPAUSE_ALL_ROLE));
+    });
+
+    it("should fail updateUnpauseTypeRole if roles are not different", async () => {
+      const updateCall = pauseManager.connect(pauseManagerAccount).updatePauseTypeRole(GENERAL_PAUSE_TYPE, PAUSE_ALL_ROLE);
+      await expectRevertWithCustomError(pauseManager, updateCall,"RolesNotDifferent");
+    });
+
+    it("should fail updateUnpauseTypeRole if roles are not different", async () => {
+      const updateCall = pauseManager.connect(pauseManagerAccount).updateUnpauseTypeRole(GENERAL_PAUSE_TYPE, UNPAUSE_ALL_ROLE);
+      await expectRevertWithCustomError(pauseManager, updateCall,"RolesNotDifferent");
+    });
+
```

Comment:
We could add test cases for 
- "should fail update pause type role with old role"
- "should fail update unPause type role with old role"